cmd/mesh/rpc_handler.go

```diff
@@ -86,8 +110,19 @@ func (handler *rpcHandler) AddOrders(signedOrdersRaw []*json.RawMessage, opts rp
 }
 
 // AddPeer is called when an RPC client calls AddPeer,
-func (handler *rpcHandler) AddPeer(peerInfo peerstore.PeerInfo) error {
+func (handler *rpcHandler) AddPeer(peerInfo peerstore.PeerInfo) (errRes error) {
 	log.Debug("received AddPeer request via RPC")
+	// Catch panics, log stack trace and return RPC error message
+	defer func() {
+		if err := recover(); err != nil {
+			log.WithFields(log.Fields{
+				"error":      err,
+				"method":     "AddPeer",
+				"stackTrace": getStackTrace(),
```

Comment:
I think you can just use [`debug.Stack`](https://golang.org/pkg/runtime/debug/#Stack) here:

```suggestion
				"stackTrace": string(debug.Stack()),
```